BUILD.bazel

```diff
@@ -194,6 +194,18 @@ cc_binary(
     deps = ["//src/google/protobuf/compiler:protoc_lib"],
 )
 
+proto_toolchain(
+    name = "protoc_source_toolchain",
```

Comment:
The `proto_toolchain` macro defined in rules_proto is causing problems with Bazel 6, because `allowlist_different_package` wasn't defined yet.

I marked https://github.com/bazelbuild/bazel/commit/6ede08f4a4243e4813b04169f24e07ba7354ef63 for cherry-picking.

What we could do in the mean time, is use https://github.com/bazel-contrib/bazel_features in rules_proto to make https://github.com/bazelbuild/rules_proto/blob/main/proto/private/rules/proto_toolchain_rule.bzl#L38 only be added on Bazel versions > 6.4.0.